backend/src/main/java/com/example/subjecthub/controller/SubjectServiceController.java

```diff
@@ -69,4 +77,88 @@ public Subject getSubject(
         // TODO: Don't allow cross university fetching
         return subjectRepository.findOne(subjectId);
     }
+
+    @Override
+    public List<SubjectComment> getCommentsByUser(
+        @PathVariable Long universityId,
+        @PathVariable Long userId
+    ){
+        return subjectCommentRepository.findByUser_Id(userId);
+    }
+
+    //TODO: ALL COMMENT IMPLEMENTATIONS: as above, reflect null on no subject, no cross-uni fetch
+    @Override
+    public List<SubjectComment> getComments(
+        @PathVariable Long universityId,
+        @PathVariable Long subjectId
+    ){
+        return subjectCommentRepository.findBySubject_Id(subjectId);
+    }
+
+    @Override
+    public SubjectComment getComment(
+        @PathVariable Long universityId,
+        @PathVariable Long subjectId,
+        @PathVariable Long commentId
+    ){
+        return subjectCommentRepository.findBySubject_IdAndId(subjectId,commentId);
+    }
+
+    @Override
+    public SubjectComment commentAdd(
+        @PathVariable Long universityId,
+        @PathVariable Long subjectId,
+        @RequestBody AddCommentRequest addCommentRequest
+    ){
+        SubjectComment newComment = new SubjectComment();
+        newComment.setPost(addCommentRequest.getComment());
+        newComment.setUser(subjectHubUserRepository.findOne(addCommentRequest.getUserId()));
```

Comment:
This line can fail to retrieve a user.

The username shouldn't be sent along with the addCommentRequest, but instead provided implicitly using Spring's in built security.

I'd advise adding a method like getRequestingUser() that returns the user who sent the request to the Controller and use that for all your POST requests. You should basically just copy and paste what I have inside the self method in the AuthController.

I'll make changes in the future to make getting the requesting user easier.

Reply:
I've tried to respond to this and researched using optional and managing nulls, not sure if I've handled it correctly.

Reply:
had to change it to work with tests, looking problematic for the future, but working- I know it needs to be improved but don't know how.